kartothek/io_components/metapartition.py

```diff
@@ -1568,24 +1575,46 @@ def parse_input_to_metapartition(obj, metadata_version=None):
             return MetaPartition(label=None, metadata_version=metadata_version)
         first_element = obj.pop()
         mp = parse_input_to_metapartition(
-            obj=first_element, metadata_version=metadata_version
+            obj=first_element,
+            metadata_version=metadata_version,
+            expected_secondary_indices=expected_secondary_indices,
         )
         for mp_in in obj:
             mp = mp.add_metapartition(
                 parse_input_to_metapartition(
-                    obj=mp_in, metadata_version=metadata_version
+                    obj=mp_in,
+                    metadata_version=metadata_version,
+                    expected_secondary_indices=expected_secondary_indices,
                 )
             )
     elif isinstance(obj, dict):
         if isinstance(obj["data"], list):
             data = dict(obj["data"])
         else:
             data = obj["data"]
+
+        indices = obj.get("indices", {})
+        if indices:
+            warnings.warn(
+                "The explicit input of indices using the `indices` key is deprecated."
+                'Use the `secondary_indices` keyword argument of "write" and "update" functions instead.',
+                DeprecationWarning,
+            )
+        if isinstance(expected_secondary_indices, set):
```

Comment:
we should at least accept lists as well. While a `set` is mathematically speaking correct for the input, I'd argue that `list`s are a more common data type and I wouldn't want to restrict the input to `set`s. That makes the interface a bit cumbersome.

Reply:
`list`s (and tuples) are normalized to `set`s after the latest commit. We can discuss this later in more detail

Reply:
This isn't normalized but explicit input, isn't it?

Reply:
You could make this section a bit easier top level and push the complexity down below

```
if expected_secondary_indices is not None:
    _ensure_valid_indices(secondary_indices=expected_secondary_indices, mp_indices=indices)

def _ensure_valid_indices():
    if secondary indices is None and mp_indices is not None:
        raise Exception
    else:
        # either case the secondary indices to a set or assert it to be a set
        if secondary_indices.issuperset(mp_indices.keys()):
            # perform validation
```
I would generally assume that a function should verify it's input values not the other way round.

Reply:
Thanks